src/Storages/ObjectStorageQueue/ObjectStorageQueueOrderedFileMetadata.cpp

```diff
@@ -446,7 +524,40 @@ void ObjectStorageQueueOrderedFileMetadata::prepareProcessedRequests(
                 "File ({}) is already processed, while expected it not to be (path: {})",
                 path, processed_node_path_);
         }
-        requests.push_back(zkutil::makeSetRequest(processed_node_path_, node_metadata.toString(), processed_node_stat.version));
+        if (created_nodes)
+        {
+            auto cn = created_nodes->find(processed_node_path_);
+            if (cn == created_nodes->end())
+            {
+                created_nodes->insert(std::make_pair(processed_node_path_, LastProcessedFileInfo({path, requests.size()})));
```

Comment:
I guess this should work
```suggestion
                created_nodes->emplace(processed_node_path_, LastProcessedFileInfo({path, requests.size()}));
```